main.py

```diff
@@ -31,11 +38,194 @@ def setup(self):
 
         So, if you have any setup routine, insert it here.
         """
+        self.basic_pipeline = BASIC_PIPELINE
+        self.subscribed_napps = SUBSCRIBED_NAPPS
         self.pipeline_controller = self.get_pipeline_controller()
+        self._pipeline_lock = Lock()
```

Comment:
You can also nuke the `self._pipeline_lock`, since we have DB guarantees when trying to enable a pipeline we won't need it, they way you're using `build_content` and passing the dict payload is thead-safe (it won't result in dict size changed during iteration)

Reply:
Deleted.